rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -361,13 +374,30 @@ def _user_intent_from_step(self, step: Dict[Text, Any]) -> Text:
             )
             # Remove leading slash
             user_intent = user_intent[1:]
-        return user_intent
+
+        # StoryStep should never contain a full retrieval intent, only the base intent.
+        # However, users can specify full retrieval intents in their test stories file
+        # for the NLU testing purposes.
+        base_intent, response_key = Message.separate_intent_response_key(user_intent)
+        if response_key and not self.is_test_stories_file(self.source_name):
+            rasa.shared.utils.io.raise_warning(
+                f"Issue found in '{self.source_name}':\n"
+                f"User intent '{user_intent}' is a retrieval intent. "
```

Comment:
this error should be slightly more specific, e.g. we have the file in there but the  user would need to search the whole file to find the issue